buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -120,4 +132,61 @@ public static void maybeConfigure(TaskContainer tasks, String name, Action<? sup
 
         return (Provider<T>) registration.getService();
     }
+
+    /**
+     * Add a source set and task of the same name that runs tests.
+     *
+     * IDEs are also configured if setup, and the test task is added to check. The new test source
+     * set extends from the normal test source set to allow sharing of utilities.
+     */
+    public static void addTestSourceSet(Project project, String sourceSetName) {
+        project.getPluginManager().apply(ElasticsearchJavaPlugin.class);
+
+        // create our test source set and task
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet extraTestSourceSet = sourceSets.create(sourceSetName);
+        TaskProvider<Test> testTask = project.getTasks().register(sourceSetName, Test.class);
+        testTask.configure(task -> {
+            task.setTestClassesDirs(extraTestSourceSet.getOutput().getClassesDirs());
+            task.setClasspath(extraTestSourceSet.getRuntimeClasspath());
```

Comment:
should this `task.setGroup(JavaBasePlugin.VERIFICATION_GROUP);` ? (It helps with the Intellij Gradle view)

Reply:
good call, done